From f4f8fb9237797044afdb473ea45097382fb1ecca Mon Sep 17 00:00:00 2001
From: RhysB <jetpackingwolf@gmail.com>
Date: Sat, 5 Oct 2019 18:04:26 +1000
Subject: [PATCH] Fix Instamine Dupes


diff --git a/src/main/java/net/minecraft/server/BlockChest.java b/src/main/java/net/minecraft/server/BlockChest.java
index 220495cb6..a4879897c 100644
--- a/src/main/java/net/minecraft/server/BlockChest.java
+++ b/src/main/java/net/minecraft/server/BlockChest.java
@@ -68,6 +68,7 @@ public class BlockChest extends BlockContainer {
                     entityitem.motZ = (double) ((float) this.a.nextGaussian() * f3);
                     world.addEntity(entityitem);
                 }
+                tileentitychest.setItem(l, null); //Canyon - Instamine Dupe Fix
             }
         }
 
diff --git a/src/main/java/net/minecraft/server/BlockDispenser.java b/src/main/java/net/minecraft/server/BlockDispenser.java
index 89975dd8d..ff8348423 100644
--- a/src/main/java/net/minecraft/server/BlockDispenser.java
+++ b/src/main/java/net/minecraft/server/BlockDispenser.java
@@ -239,6 +239,7 @@ public class BlockDispenser extends BlockContainer {
                     entityitem.motZ = (double) ((float) this.a.nextGaussian() * f3);
                     world.addEntity(entityitem);
                 }
+                tileentitydispenser.setItem(l, null); //Canyon - Instamine Dupe Fix
             }
         }
 
diff --git a/src/main/java/net/minecraft/server/BlockFurnace.java b/src/main/java/net/minecraft/server/BlockFurnace.java
index ab9f6bd96..6b003e3d4 100644
--- a/src/main/java/net/minecraft/server/BlockFurnace.java
+++ b/src/main/java/net/minecraft/server/BlockFurnace.java
@@ -137,6 +137,7 @@ public class BlockFurnace extends BlockContainer {
                         entityitem.motZ = (double) ((float) this.a.nextGaussian() * f3);
                         world.addEntity(entityitem);
                     }
+                    tileentityfurnace.setItem(l, null); //Canyon - Instamine Dupe Fix
                 }
             }
         }
diff --git a/src/main/java/net/minecraft/server/EntityMinecart.java b/src/main/java/net/minecraft/server/EntityMinecart.java
index 60e1bdf01..32c583fa2 100644
--- a/src/main/java/net/minecraft/server/EntityMinecart.java
+++ b/src/main/java/net/minecraft/server/EntityMinecart.java
@@ -160,6 +160,7 @@ public class EntityMinecart extends Entity implements IInventory {
                                 entityitem.motZ = (double) ((float) this.random.nextGaussian() * f3);
                                 this.world.addEntity(entityitem);
                             }
+                            entityminecart.setItem(j, null); //Canyon - Instamine Dupe Fix
                         }
                     }
 
-- 
2.33.0

