From 4a02818b5b40fed833fc693c7e5fdfbf9d355a0f Mon Sep 17 00:00:00 2001
From: Andrew Steinborn <git@steinborn.me>
Date: Mon, 30 Sep 2019 02:28:06 -0400
Subject: [PATCH] User-configurable autosave


diff --git a/src/main/java/com/canyonmodded/config/CanyonConfig.java b/src/main/java/com/canyonmodded/config/CanyonConfig.java
index 971ac06c..fbd18a31 100644
--- a/src/main/java/com/canyonmodded/config/CanyonConfig.java
+++ b/src/main/java/com/canyonmodded/config/CanyonConfig.java
@@ -7,6 +7,7 @@ import java.io.File;
 public class CanyonConfig {
     // Add settings here. Make sure they're public visibility!
     public static boolean useNewNetworkingStack = false;
+    public static int autoSave = 6000;
 
     public static void loadConfiguration(File file) {
         Configuration configuration = new Configuration(file);
@@ -14,6 +15,7 @@ public class CanyonConfig {
 
         // Load settings here.
         useNewNetworkingStack = configuration.getBoolean("settings.use-new-networking-stack", false);
+        autoSave = configuration.getInt("settings.auto-save-every", 6000);
 
         configuration.save();
     }
diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 4c19b963..1d4ad166 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1728,7 +1728,7 @@ public class World implements IBlockAccess {
         }
 
         i = this.worldData.f() + 1L;
-        if (i % (long) this.p == 0L) {
+        if (MinecraftServer.currentTick % com.canyonmodded.config.CanyonConfig.autoSave == 0L) { // Canyon - user-configurable autosave
             this.save(false, (IProgressUpdate) null);
         }
 
-- 
2.22.0.windows.1

