From 60e9199236ef93fd0f537a1eb976b2e151e65e96 Mon Sep 17 00:00:00 2001
From: RhysB <jetpackingwolf@gmail.com>
Date: Fri, 4 Oct 2019 23:41:14 +1000
Subject: [PATCH] Fix Dupe Glitches Including Instamine


diff --git a/src/main/java/net/minecraft/server/BlockChest.java b/src/main/java/net/minecraft/server/BlockChest.java
new file mode 100644
index 00000000..be79eb2b
--- /dev/null
+++ b/src/main/java/net/minecraft/server/BlockChest.java
@@ -0,0 +1,120 @@
+package net.minecraft.server;
+
+import java.util.Random;
+
+public class BlockChest extends BlockContainer {
+
+    private Random a = new Random();
+
+    protected BlockChest(int i) {
+        super(i, Material.WOOD);
+        this.textureId = 26;
+    }
+
+    public int a(int i) {
+        return i == 1 ? this.textureId - 1 : (i == 0 ? this.textureId - 1 : (i == 3 ? this.textureId + 1 : this.textureId));
+    }
+
+    public boolean canPlace(World world, int i, int j, int k) {
+        int l = 0;
+
+        if (world.getTypeId(i - 1, j, k) == this.id) {
+            ++l;
+        }
+
+        if (world.getTypeId(i + 1, j, k) == this.id) {
+            ++l;
+        }
+
+        if (world.getTypeId(i, j, k - 1) == this.id) {
+            ++l;
+        }
+
+        if (world.getTypeId(i, j, k + 1) == this.id) {
+            ++l;
+        }
+
+        return l > 1 ? false : (this.g(world, i - 1, j, k) ? false : (this.g(world, i + 1, j, k) ? false : (this.g(world, i, j, k - 1) ? false : !this.g(world, i, j, k + 1))));
+    }
+
+    private boolean g(World world, int i, int j, int k) {
+        return world.getTypeId(i, j, k) != this.id ? false : (world.getTypeId(i - 1, j, k) == this.id ? true : (world.getTypeId(i + 1, j, k) == this.id ? true : (world.getTypeId(i, j, k - 1) == this.id ? true : world.getTypeId(i, j, k + 1) == this.id)));
+    }
+
+    public void remove(World world, int i, int j, int k) {
+        TileEntityChest tileentitychest = (TileEntityChest) world.getTileEntity(i, j, k);
+
+        for (int l = 0; l < tileentitychest.getSize(); ++l) {
+            ItemStack itemstack = tileentitychest.getItem(l);
+
+            if (itemstack != null) {
+                float f = this.a.nextFloat() * 0.8F + 0.1F;
+                float f1 = this.a.nextFloat() * 0.8F + 0.1F;
+                float f2 = this.a.nextFloat() * 0.8F + 0.1F;
+
+                while (itemstack.count > 0) {
+                    int i1 = this.a.nextInt(21) + 10;
+
+                    if (i1 > itemstack.count) {
+                        i1 = itemstack.count;
+                    }
+
+                    itemstack.count -= i1;
+                    EntityItem entityitem = new EntityItem(world, (double) ((float) i + f), (double) ((float) j + f1), (double) ((float) k + f2), new ItemStack(itemstack.id, i1, itemstack.getData()));
+                    float f3 = 0.05F;
+
+                    entityitem.motX = (double) ((float) this.a.nextGaussian() * f3);
+                    entityitem.motY = (double) ((float) this.a.nextGaussian() * f3 + 0.2F);
+                    entityitem.motZ = (double) ((float) this.a.nextGaussian() * f3);
+                    world.addEntity(entityitem);
+                }
+                tileentitychest.setItem(l, null);
+            }
+        }
+
+        super.remove(world, i, j, k);
+    }
+
+    public boolean interact(World world, int i, int j, int k, EntityHuman entityhuman) {
+        Object object = (TileEntityChest) world.getTileEntity(i, j, k);
+
+        if (world.e(i, j + 1, k)) {
+            return true;
+        } else if (world.getTypeId(i - 1, j, k) == this.id && world.e(i - 1, j + 1, k)) {
+            return true;
+        } else if (world.getTypeId(i + 1, j, k) == this.id && world.e(i + 1, j + 1, k)) {
+            return true;
+        } else if (world.getTypeId(i, j, k - 1) == this.id && world.e(i, j + 1, k - 1)) {
+            return true;
+        } else if (world.getTypeId(i, j, k + 1) == this.id && world.e(i, j + 1, k + 1)) {
+            return true;
+        } else {
+            if (world.getTypeId(i - 1, j, k) == this.id) {
+                object = new InventoryLargeChest("Large chest", (TileEntityChest) world.getTileEntity(i - 1, j, k), (IInventory) object);
+            }
+
+            if (world.getTypeId(i + 1, j, k) == this.id) {
+                object = new InventoryLargeChest("Large chest", (IInventory) object, (TileEntityChest) world.getTileEntity(i + 1, j, k));
+            }
+
+            if (world.getTypeId(i, j, k - 1) == this.id) {
+                object = new InventoryLargeChest("Large chest", (TileEntityChest) world.getTileEntity(i, j, k - 1), (IInventory) object);
+            }
+
+            if (world.getTypeId(i, j, k + 1) == this.id) {
+                object = new InventoryLargeChest("Large chest", (IInventory) object, (TileEntityChest) world.getTileEntity(i, j, k + 1));
+            }
+
+            if (world.isStatic) {
+                return true;
+            } else {
+                entityhuman.a((IInventory) object);
+                return true;
+            }
+        }
+    }
+
+    protected TileEntity a_() {
+        return new TileEntityChest();
+    }
+}
diff --git a/src/main/java/net/minecraft/server/BlockDispenser.java b/src/main/java/net/minecraft/server/BlockDispenser.java
index 89975dd8..bf54c6cf 100644
--- a/src/main/java/net/minecraft/server/BlockDispenser.java
+++ b/src/main/java/net/minecraft/server/BlockDispenser.java
@@ -214,6 +214,9 @@ public class BlockDispenser extends BlockContainer {
 
     public void remove(World world, int i, int j, int k) {
         TileEntityDispenser tileentitydispenser = (TileEntityDispenser) world.getTileEntity(i, j, k);
+        if( tileentitydispenser == null) {
+            return;
+        }
 
         for (int l = 0; l < tileentitydispenser.getSize(); ++l) {
             ItemStack itemstack = tileentitydispenser.getItem(l);
@@ -239,6 +242,7 @@ public class BlockDispenser extends BlockContainer {
                     entityitem.motZ = (double) ((float) this.a.nextGaussian() * f3);
                     world.addEntity(entityitem);
                 }
+                tileentitydispenser.setItem(l, null);
             }
         }
 
diff --git a/src/main/java/net/minecraft/server/BlockFurnace.java b/src/main/java/net/minecraft/server/BlockFurnace.java
index ab9f6bd9..5c306e59 100644
--- a/src/main/java/net/minecraft/server/BlockFurnace.java
+++ b/src/main/java/net/minecraft/server/BlockFurnace.java
@@ -137,6 +137,7 @@ public class BlockFurnace extends BlockContainer {
                         entityitem.motZ = (double) ((float) this.a.nextGaussian() * f3);
                         world.addEntity(entityitem);
                     }
+                    tileentityfurnace.setItem(l, null);
                 }
             }
         }
diff --git a/src/main/java/net/minecraft/server/EntityMinecart.java b/src/main/java/net/minecraft/server/EntityMinecart.java
index 60e1bdf0..85d7563e 100644
--- a/src/main/java/net/minecraft/server/EntityMinecart.java
+++ b/src/main/java/net/minecraft/server/EntityMinecart.java
@@ -160,6 +160,7 @@ public class EntityMinecart extends Entity implements IInventory {
                                 entityitem.motZ = (double) ((float) this.random.nextGaussian() * f3);
                                 this.world.addEntity(entityitem);
                             }
+                            entityminecart.setItem(j, null);
                         }
                     }
 
-- 
2.17.1

