From 3e79447562a7142f040dba457f478fbba09de920 Mon Sep 17 00:00:00 2001
From: Egor <egor@opensrc.club>
Date: Tue, 7 Sep 2021 02:28:54 +0300
Subject: [PATCH] fix creeper music disc drops


diff --git a/src/main/java/net/minecraft/server/EntityArrow.java b/src/main/java/net/minecraft/server/EntityArrow.java
index 614b2955e..f533487ba 100644
--- a/src/main/java/net/minecraft/server/EntityArrow.java
+++ b/src/main/java/net/minecraft/server/EntityArrow.java
@@ -187,7 +187,7 @@ public class EntityArrow extends Entity {
                             stick = !projectile.doesBounce();
                         } else {
                             // this function returns if the arrow should stick in or not, i.e. !bounce
-                            stick = movingobjectposition.entity.damageEntity(this, event.getDamage());
+                            stick = movingobjectposition.entity.damageEntity(this.shooter, event.getDamage()); // Canyon - use shooter as damage source
                         }
                     } else {
                         stick = movingobjectposition.entity.damageEntity(this.shooter, 4);
-- 
2.33.0

