From 64e2d1a7859057c4e92581e19093f2796ec7f36b Mon Sep 17 00:00:00 2001
From: KoboDev <kobodev@mail.com>
Date: Thu, 15 Apr 2021 16:00:45 -0500
Subject: [PATCH] hopefully this works


diff --git a/pom.xml b/pom.xml
index b0c60311..a4e39215 100644
--- a/pom.xml
+++ b/pom.xml
@@ -24,7 +24,7 @@
   <repositories>
     <repository>
       <id>bukkit-repo</id>
-      <url>https://download938.mediafire.com/jzs3lz3gm1jg/xuiz2lwjyfiljfp/minecraft-server.jar</url> <!-- Canyon: replace repo -->
+      <url>https://github.com/KoboDev/SupplyAndDemand/releases/download/BukkitRepo/minecraft-server.jar</url> <!-- Canyon: replace repo -->
     </repository>
     <repository>
       <id>canyon-local-repo</id>
@@ -63,14 +63,14 @@
     <dependency>
       <groupId>org.jline</groupId>
       <artifactId>jline</artifactId>
-      <version>3.14.0</version>
+      <version>3.14.1</version>
       <type>jar</type>
       <scope>compile</scope>
     </dependency>
     <dependency>
       <groupId>org.jline</groupId>
       <artifactId>jline-terminal-jansi</artifactId>
-      <version>3.14.0</version>
+      <version>3.14.1</version>
       <type>jar</type>
       <scope>compile</scope>
     </dependency>
diff --git a/src/main/java/net/minecraft/server/ChunkRegionLoader.java b/src/main/java/net/minecraft/server/ChunkRegionLoader.java
index c4401e9a..68279b79 100644
--- a/src/main/java/net/minecraft/server/ChunkRegionLoader.java
+++ b/src/main/java/net/minecraft/server/ChunkRegionLoader.java
@@ -5,6 +5,7 @@ import java.io.DataInputStream;
 import java.io.DataOutput;
 import java.io.DataOutputStream;
 import java.io.File;
+import java.io.IOException;
 
 public class ChunkRegionLoader implements IChunkLoader {
 
@@ -18,7 +19,13 @@ public class ChunkRegionLoader implements IChunkLoader {
         DataInputStream datainputstream = RegionFileCache.c(this.a, i, j);
 
         if (datainputstream != null) {
-            NBTTagCompound nbttagcompound = CompressedStreamTools.a((DataInput) datainputstream);
+            NBTTagCompound nbttagcompound = null;
+			try {
+				nbttagcompound = CompressedStreamTools.a((DataInput) datainputstream);
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
 
             if (!nbttagcompound.hasKey("Level")) {
                 System.out.println("Chunk file at " + i + "," + j + " is missing level data, skipping");
diff --git a/src/main/java/net/minecraft/server/CompressedStreamTools.java b/src/main/java/net/minecraft/server/CompressedStreamTools.java
index 044e3566..454968ad 100644
--- a/src/main/java/net/minecraft/server/CompressedStreamTools.java
+++ b/src/main/java/net/minecraft/server/CompressedStreamTools.java
@@ -23,10 +23,15 @@ public class CompressedStreamTools {
 			e.printStackTrace();
 		}
 
-        NBTTagCompound nbttagcompound;
+        NBTTagCompound nbttagcompound = null;
 
         try {
-            nbttagcompound = a((DataInput) datainputstream);
+            try {
+				nbttagcompound = a((DataInput) datainputstream);
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
         } finally {
             try {
 				datainputstream.close();
@@ -60,7 +65,7 @@ public class CompressedStreamTools {
         }
     }
 
-    public static NBTTagCompound a(DataInput datainput) {
+    public static NBTTagCompound a(DataInput datainput) throws IOException {
         NBTBase nbtbase = NBTBase.b(datainput);
 
         if (nbtbase instanceof NBTTagCompound) {
-- 
2.31.1.windows.1

